sdk/storage/azure-storage-file-share/tests/test_share_async.py

```diff
@@ -1650,3 +1650,51 @@ async def test_delete_directory_from_share(self, **kwargs):
         assert len(resp) == 2
         await self._delete_shares(share.share_name)
 
+    @FileSharePreparer()
+    @recorded_by_proxy_async
+    async def test_share_paid_bursting(self, **kwargs):
+        premium_storage_file_account_name = kwargs.pop("premium_storage_file_account_name")
+        premium_storage_file_account_key = kwargs.pop("premium_storage_file_account_key")
+
+        # Arrange
+        self._setup(premium_storage_file_account_name, premium_storage_file_account_key)
+        max_mibps = 10340
+        max_iops = 102400
+
+        # Act / Assert
+        share = await self._create_share(
+            paid_bursting_enabled=True,
+            paid_bursting_max_bandwidth_mibps=5000,
+            paid_bursting_max_iops=1000
+        )
+        share_props = await share.get_share_properties()
+        assert share_props.paid_bursting_enabled
+        assert share_props.paid_bursting_max_bandwidth_mibps == 5000
+        assert share_props.paid_bursting_max_iops == 1000
+
+        await share.set_share_properties(
+            root_squash="NoRootSquash",
+            paid_bursting_enabled=True,
+            paid_bursting_max_bandwidth_mibps=max_mibps,
+            paid_bursting_max_iops=max_iops
+        )
+        share_props = await share.get_share_properties()
+        assert share_props.paid_bursting_enabled
+        assert share_props.paid_bursting_max_bandwidth_mibps == max_mibps
+        assert share_props.paid_bursting_max_iops == max_iops
+
+        shares = None
+        async for share in self.fsc.list_shares():
+            if shares is None:
+                shares = []
```

Comment:
Sorry I didn't see this before but why not start `shares = []` on line 1686 and then remove the assert on line 1692? Then you could remove this if check from every iteration. 

This is a nitpick for a test so you don't have to fix it but thought I would just suggest it :).

Reply:
Initially I followed the existing test patterns of asserting not `None`, but I think this is a cleaner way to write it :) Will update the test as well as the recordings!